[ymirmeddeb] IP

ymirmeddeb opened the pull request.

Commit:
changed the name of the chatbot and the message it prints out.

Commit:
added echo functionality

Commit:
added functionality to add tasks to a list and display them

rexyyong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String logo = " ╔╦═╦══╦═╦╗\n" +
```
I like that you used a logo for your program name, it is cool!

rexyyong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+        System.out.println("Hiiii, my name is\n" + logo );
+        System.out.println("I am here to help :3\n");
+        String line;
+        String[] tasks = new String[100];
```
May I clarify if there is a reason why a Task Class was not created to store the array of tasks?

rexyyong replied:
Consider using a constant for maximum amount of tasks. Otherwise readers may be confused why you assigned an array of size 100 in the future. A constant makes it more intuitive that the number did come up randomly.

Jai2501 replied:
Avoid the use of magic numbers. Either assign this number to a constant and use that, or you may use an `ArrayList` in which you do not have to declare the size during the initialisation, and it takes care of that dynamically internally.

rexyyong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+                nextTask++;
+                System.out.println("I just added " + line + " to your list!!!");
+            }
+            System.out.print("Yes darling, what can i do for you?\n");
```
I like that you included this prompt for the user after they have keyed in the task or keyed in list. I feel that the prompt is a good way to remind users that they can continue putting in inputs

rexyyong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+        System.out.println("I am here to help :3\n");
+        String line;
+        String[] tasks = new String[100];
+        int nextTask = 0;
```
An alternative way to name nextTask would be taskNumber of totalTasks.

rexyyong reviewed (commented):
Overall the code was easy to read. However, I could not find your level-4, level-3 code.

Kishen271828 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+    public static void echo(String word){
+        System.out.println(word);
+        return;
+    }
```
The echo method is defined but not used in the main method or anywhere else in the class. Consider removing unused methods to declutter the code.

Jai2501 replied:
Unused methods/variables should be removed to improve readability.

Kishen271828 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+                tasks[nextTask] = line;
+                nextTask++;
+                System.out.println("I just added " + line + " to your list!!!");
+            }
```
The 'list' condition is checked twice unnecessarily. Consider replacing it with an if-else block
```suggestion
            if (line.equals("list")) {
                printTasks(tasks, nextTask);
            } else {
                tasks[nextTask] = line;
                nextTask++;
                System.out.println("I just added " + line + " to your list!!!");
            }
```

Jai2501 replied:
Another way of doing this would be as follows. This would improve the readability of your code.

```Java
while (!line.equals("bye")){
    if (line.equals("list")){
        printTasks(tasks, nextTask);
        continue;
    }

    tasks[nextTask] = line;
    nextTask++;
    System.out.println("I just added " + line + " to your list!!!");
    System.out.print("Yes darling, what can i do for you?\n");
    line = in.nextLine();
}

 System.out.println("Love ya <3 See ya!!\n");
```

Kishen271828 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+        int nextTask = 0;
+        Scanner in = new Scanner(System.in);
+        System.out.print("Yes darling, what can i do for you?\n");
+        line = in.nextLine();
```
The code here could be structured as follows to improve the logical structure. Also avoid using magic numbers.
```suggestion
        String[] tasks = new String[MAX_TASKS];
        int nextTask = 0;
        System.out.print("Yes darling, what can i do for you?\n");
        
        String line;
        Scanner in = new Scanner(System.in);
        line = in.nextLine();
```

Kishen271828 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+    }
+
+    public static void printTasks(String[] tasks, int nextTask){
+        if (tasks[0] == null){
+            System.out.println("Your list is empty ;_;");
+            return;
+        }
```
Well done on handling the case when there are no tasks!

Kishen271828 reviewed (commented):
Overall the code LGTM. There are a few instances where the code quality could be improved. Consider using line spacing to improve readability.

Commit:
added mark and unmark functionality

Commit:
added todo, deadline and event functionality

Commit:
implemented testing

Jai2501 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
+
+}
```
Missing EOF. Add an extra empty line at the end of file (EOF), as this indicates the end of the file. This has been observed in multiple files.

You may configure IntelliJ to add this automatically for you when hit `CTRL/CMD` + `S`

Jai2501 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+        super(desc);
+        this.from = from;
+        this.to = to;
+    }
```
Add a line break between logically linked statements. This helps improve readability as well.

```Java 
protected String to;
    
public Event(String desc, String from, String to) {
    super(desc);
    this.from = from;
    this.to = to;
}
```

Jai2501 commented on src/main/java/Joe.java:
```diff
@@ -0,0 +1,114 @@
+                printHelp();
+            }
+            else if (userInput.startsWith("unmark ")){
+                unmarkTask(tasks, userInput, taskCount);
+                printHelp();
+            }
+            else if (userInput.startsWith("todo ")){
+                ToDo userToDo = new ToDo(userInput.replace("todo ", "").trim());
+                tasks[taskCount] = userToDo;
+                taskCount++;
+                System.out.println("Alrightyy, im gonna add " + userInput.replace("todo ", "").trim() + " to your ToDo list!!!");
+                printHelp();
+            }
+            else if (userInput.startsWith("deadline ")){
+                int due = userInput.indexOf("/by");
+                String by = userInput.substring(due + 3).trim();
+                String deadline = userInput.substring(8, due).trim();
+                Deadline userDeadline = new Deadline(deadline, by);
+                tasks[taskCount] = userDeadline;
+                taskCount++;
+                System.out.println("Alrightyy, im gonna add " + deadline + " by " + by +" to your deadlines!!!");
+                printHelp();
+            }
+            else if (userInput.startsWith("event ")) {
+                int splitFrom = userInput.indexOf("/from");
+                int splitTo = userInput.indexOf("/to");
+                String from = userInput.substring(splitFrom + 5, splitTo).trim();
+                String to = userInput.substring(splitTo + 3).trim();
+                String event = userInput.substring(5, splitFrom).trim();
+                Event userEvent = new Event(event, from, to);
+                tasks[taskCount] = userEvent;
+                taskCount++;
+                System.out.println("Alrightyy, im gonna add " + event + " from " + from + " to " + to + " to your events!!!");
+                printHelp();
+            }
+            else {
+                System.out.println("Babe come on, what am i gonna do with this???? For the love of God, please give me a valid command.");
+                printHelp();
+            }
+        }
```
This needs to be refactored:
1. This is a very long method
2. There is a lot of nesting
3. There is absence of abstraction

Please work on it.

Jai2501 commented on src/main/java/Joe.java:
```diff
@@ -0,0 +1,114 @@
+
+public class Joe {
+    public static void main(String[] args) {
+        int maxTasks = 100;
+        Task[] tasks = new Task[100];
```
Same, avoid use of Magic Numbers.

Jai2501 reviewed (commented):
Hi Ymir!

Good job on the work done so far!

I have reviewed your code and have left some comments. Please go through them and get back to me for any clarifications.

Thanks!

Commit:
added error handling

Commit:
no message

Commit:
no message

Commit:
no message

Commit:
Revert "no message"

This reverts commit fcf12c01e94622965c9a6d930be4ce4069aad3c5.

Commit:
added error handling

Commit:
no message

Commit:
Merge branch 'master' of https://github.com/ymirmeddeb/ip

* 'master' of https://github.com/ymirmeddeb/ip:
  no message

Commit:
added delete functionality

Commit:
Merge branch 'branch-level-6'

* branch-level-6:
  added delete functionality

Commit:
added text reading and writing functionality

Commit:
added more OOP

Commit:
implemented find functionality

Commit:
no message

ymirmeddeb committed:
Merge pull request #1 from ymirmeddeb/branch-level-9

Level 9

Commit:
added save functionality

Commit:
added save functionality

Commit:
no message

Commit:
no message

ymirmeddeb committed:
Merge pull request #2 from ymirmeddeb/branch-level-7

Branch level 7

ymirmeddeb committed:
Update README.md

ymirmeddeb committed:
Update Joe.java

updated this because my last commit was the wrong code due to some confusion and problems with using Sourcetree

ymirmeddeb committed:
Update Deadline.java

updated this because my last commit was the wrong code due to some confusion and problems with using Sourcetree

ymirmeddeb committed:
Update Parser.java

updated this because my last commit was the wrong code due to some confusion and problems with using Sourcetree

ymirmeddeb committed:
Update Storage.java

updated this because my last commit was the wrong code due to some confusion and problems with using Sourcetree

ymirmeddeb committed:
A-UserGuide

made the UG
